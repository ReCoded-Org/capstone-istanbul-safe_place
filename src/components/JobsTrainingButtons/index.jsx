import React, { useState } from "react";
import { useTranslation } from "react-i18next";
import "./index.scss";

const BUTTON_SPECS = [
  {
    contentTranslationKey: "home.jobOpportunities",
    buttonTranslationKey: "jobsTrainingPage.buttons.jobOpportunities",
  },
  {
    contentTranslationKey: "home.trainings",
    buttonTranslationKey: "jobsTrainingPage.buttons.trainings",
  },
];

const JobsTrainingButtons = ({ setCardsArr }) => {
  const { t } = useTranslation();
  const [selectedButtonIndex, setSelectedButtonIndex] = useState(0);

  // This variables maps the button returns a button for each.
  const opportunityListBtns = BUTTON_SPECS.map((button, index) => {
    const opportunityData = [
      ...t(`${button.contentTranslationKey}`, { returnObjects: true }),
    ];

    const getOpportunityData = () => {
      setCardsArr(opportunityData);
      setSelectedButtonIndex(index);
    };

    return (
      <button
        key={t(button.contentTranslationKey)}
        type="button"
        className="opportunityListBtn"
        onClick={() => {
          getOpportunityData(index);
        }}
        disabled={index === selectedButtonIndex}
      >
        {t(button.buttonTranslationKey)}
      </button>
    );
  });

  return <div>{opportunityListBtns}</div>;
};

export default JobsTrainingButtons;
