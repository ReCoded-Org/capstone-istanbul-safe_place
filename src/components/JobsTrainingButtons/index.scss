@import "../../style/variables";
@import "../../style/buttons";

.opportunityListBtn {
  @include primaryBtn;
  width: 12.75rem;
  border-radius: 0;
  text-align: right;
  color: $darkColor;
  background-color: transparent;

  &:focus,
  &:active,
  &:hover,
  &:disabled {
    border-right: solid $primaryColor;
    outline: 0;
    color: $primaryColor;
    background-color: transparent;
  }
}
