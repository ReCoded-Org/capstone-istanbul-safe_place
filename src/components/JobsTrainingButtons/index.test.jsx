import React from "react";
import renderer from "react-test-renderer";
import JobsTrainingButtons from "./index";

it("renders correctly and matches the snapshot", () => {
  const tree = renderer.create(<JobsTrainingButtons />).toJSON();
  expect(tree).toMatchSnapshot();
});
