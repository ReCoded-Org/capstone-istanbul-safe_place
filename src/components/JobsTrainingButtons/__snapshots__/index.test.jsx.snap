// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly and matches the snapshot 1`] = `
<div>
  <button
    className="opportunityListBtn"
    disabled={true}
    onClick={[Function]}
    type="button"
  >
    jobsTrainingPage.buttons.jobOpportunities
  </button>
  <button
    className="opportunityListBtn"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    jobsTrainingPage.buttons.trainings
  </button>
</div>
`;
